src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt

```diff
@@ -2,60 +2,57 @@ package uk.gov.justice.digital.hmpps.hmppstier.client
 
 import com.fasterxml.jackson.annotation.JsonCreator
 import com.fasterxml.jackson.annotation.JsonProperty
+import kotlinx.coroutines.flow.toList
 import org.springframework.beans.factory.annotation.Qualifier
-import org.springframework.core.ParameterizedTypeReference
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.awaitBody
+import org.springframework.web.reactive.function.client.bodyToFlow
 import java.math.BigDecimal
 import java.time.LocalDate
 
 @Component
 class CommunityApiClient(@Qualifier("communityWebClientAppScope") private val webClient: WebClient) {
 
-  fun getRegistrations(crn: String): Collection<Registration> =
+  suspend fun getRegistrations(crn: String): Collection<Registration> =
     webClient
       .get()
       .uri("/offenders/crn/$crn/registrations?activeOnly=true")
       .retrieve()
-      .bodyToMono(Registrations::class.java)
-      .block()?.registrations ?: listOf()
+      .awaitBody<Registrations>().registrations ?: listOf()
 
-  fun getDeliusAssessments(crn: String): DeliusAssessmentsDto? {
+  suspend fun getDeliusAssessments(crn: String): DeliusAssessmentsDto? {
     return webClient
       .get()
       .uri("/offenders/crn/$crn/assessments")
       .retrieve()
-      .bodyToMono(DeliusAssessmentsDto::class.java)
-      .block()
+      .awaitBody()
   }
 
-  fun getConvictions(crn: String): List<ConvictionDto> {
-    val responseType = object : ParameterizedTypeReference<List<ConvictionDto>>() {}
+  suspend fun getConvictions(crn: String): List<ConvictionDto> {
     return webClient
       .get()
       .uri("/offenders/crn/$crn/convictions?activeOnly=true")
       .retrieve()
-      .bodyToMono(responseType)
-      .block() ?: listOf()
+      .bodyToFlow<ConvictionDto>()
```

Comment:
I am a bit confused as to why this uses `bodyToFlow` when the one below - which previously used the same `bodyToMono` uses `awaitBody`.

Reply:
The return type `[{...},{...}]` which is in effect a Flow, the one below is `{nsis:[{...},{...}]}` which is a single object wrapping an array so needs to be an awaitBody

Reply:
Ok... oh-hum it just seems unnecessarily complicated (that's kotlin's approach not your code) - i think its trying to preserve interoperability with Java Reactive whilst adding primitives that make some things easier to do?!